test/unit/stats.js

```diff
@@ -0,0 +1,488 @@
+'use strict';
+
+const fs = require('fs');
+const assert = require('assert');
+// const exifReader = require('exif-reader');
+// const icc = require('icc');
+
+const sharp = require('../../');
+const fixtures = require('../fixtures');
+
+// Test Helpers
+var threshold = 0.001;
+function isInAcceptableRange (actual, expected) {
+  return actual >= ((1 - threshold) * expected) && actual <= ((1 + threshold) * expected);
+}
+function isInRange (actual, min, max) {
+  return actual >= min && actual <= max;
+}
+function isInteger (val) {
+  return Number.isInteger(val);
+}
+
+describe('Image Stats', function () {
+  it('JPEG', function (done) {
+    sharp(fixtures.inputJpg).stats(function (err, stats) {
+      if (err) throw err;
+
+      assert.strictEqual(true, stats.isOpaque);
+
+      // red channel
+      assert.strictEqual(0, stats.channels[0]['min']);
+      assert.strictEqual(255, stats.channels[0]['max']);
+      assert.strictEqual(true, isInAcceptableRange(stats.channels[0]['sum'], 615101275));
+      assert.strictEqual(true, isInAcceptableRange(stats.channels[0]['squaresSum'], 83061892917));
+      assert.strictEqual(true, isInAcceptableRange(stats.channels[0]['mean'], 101.44954540768993));
+      assert.strictEqual(true, isInAcceptableRange(stats.channels[0]['stdev'], 58.373870588815414));
+      assert.strictEqual(true, isInteger(stats.channels[0]['minX']) && isInRange(stats.channels[0]['minX'], 0, 2725));
```

Comment:
Perhaps split these `assert(true, x && y)` statements into two separate `assert` statements to aid debugging and remove conditional logic from tests.